projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -101,6 +101,45 @@ public function register_rest_routes() {
 				),
 			)
 		);
+
+		register_rest_route(
+			'jetpack/v4',
+			'/publicize/generate-preview-token',
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'generate_preview_token' ),
+				'permission_callback' => array( $this, 'require_admin_privilege_callback' ),
+				'args'                => array(
+					'text'      => array(
+						'description'       => __( 'The text to be used to generate the image.', 'jetpack-publicize-pkg' ),
+						'type'              => 'string',
+						'required'          => true,
+						'validate_callback' => function ( $param ) {
+							return is_string( $param );
+						},
+						'sanitize_callback' => 'sanitize_textarea_field',
+					),
+					'image_url' => array(
+						'description'       => __( 'The URL of the background image to use when generating the social image.', 'jetpack-publicize-pkg' ),
+						'type'              => 'string',
+						'required'          => false,
+						'validate_callback' => function ( $param ) {
+							return is_string( $param );
+						},
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+					'template'  => array(
+						'description'       => __( 'Array of external connection IDs to skip sharing.', 'jetpack-publicize-pkg' ),
+						'type'              => 'array',
+						'required'          => false,
+						'validate_callback' => function ( $param ) {
+							return in_array( $param, array( 'highway', 'dois', 'fullscreen', 'edge' ), true );
```

Comment:
We should be able to update this to use [the constant](https://github.com/Automattic/jetpack/pull/29596/) once #29590 is merged